File: Deep/Deep/Features/Onboarding/Store/MockAccountStore.swift
#if DEBUG
import Foundation
import Observation

/// In-memory `AccountStore` for previews and tests — no networking, no
/// persistence. Sign-up/log-in fabricate a local `Account`.
@MainActor
@Observable
final class MockAccountStore: AccountStore {
  private(set) var account: Account?
  private(set) var isRestoring = false

  init(account: Account? = nil) {
    self.account = account
  }

  func restore() async {}

  func signUp(name: String, email: String, password: String) async throws {
    let trimmedName = name.trimmingCharacters(in: .whitespacesAndNewlines)
    guard !trimmedName.isEmpty else { throw AccountError.emptyName }
    guard email.contains("@"), email.contains(".") else { throw AccountError.invalidEmail }
    guard password.count >= 8 else { throw AccountError.weakPassword }
    account = Account(displayName: trimmedName, email: email, method: .email, appleUserID: nil)
  }

  func logIn(email: String, password: String) async throws {
    account = Account(displayName: "Alex", email: email, method: .email, appleUserID: nil)
  }

  func logOut() async {
    account = nil
  }
}

extension MockAccountStore {
  static var signedOut: MockAccountStore { MockAccountStore() }

  static var appleUser: MockAccountStore {
    MockAccountStore(account: Account(displayName: "Miyu", email: nil, method: .apple, appleUserID: "x"))
  }

  static var emailUser: MockAccountStore {
    MockAccountStore(account: Account(displayName: "Alex", email: "alex@deep.app", method: .email, appleUserID: nil))
  }
}
#endif

File: android/app/src/main/java/io/appbeyond/freelance/deep/feature/onboarding/store/MockAccountStore.kt
package io.appbeyond.freelance.deep.feature.onboarding.store

import io.appbeyond.freelance.deep.auth.Account
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow

/**
 * In-memory [AccountStore] for previews — no networking, no persistence. The
 * Android twin of `MockAccountStore.swift`. Sign-up/log-in fabricate a local
 * [Account] rather than validating anything; that validation is
 * [io.appbeyond.freelance.deep.onboarding.model.SignUpValidation]'s job and is
 * exercised on its own.
 */
class MockAccountStore(account: Account? = null) : AccountStore {

  private val _account = MutableStateFlow(account)
  override val account: StateFlow<Account?> = _account.asStateFlow()

  override suspend fun restore() {}

  override suspend fun signUp(displayName: String, email: String, password: String): Account {
    val fresh = Account(id = "mock", email = email, displayName = displayName)
    _account.value = fresh
    return fresh
  }

  override suspend fun logIn(email: String, password: String): Account {
    val fresh = Account(id = "mock", email = email, displayName = "Alex")
    _account.value = fresh
    return fresh
  }

  override suspend fun logOut() {
    _account.value = null
  }

  override suspend fun deleteAccount() {
    _account.value = null
  }

  companion object {
    val signedOut: MockAccountStore get() = MockAccountStore()

    val emailUser: MockAccountStore
      get() = MockAccountStore(Account(id = "mock-alex", email = "alex@deep.app", displayName = "Alex"))
  }
}
